### PR TITLE
DEV: More bulk select modal enhancements

### PR DESCRIPTION
**Pull request by oblakeerickson**
- Add cancel button
- Add loading spinner
- Add toast

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs [hunk 1]
```diff
@@ -41,8 +43,8 @@ export default class BulkTopicActions extends Component {
       return this._processChunks(operation);
     } catch {
       this.dialog.alert(i18n("generic_error"));
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs:45]: Should this line be removed now that we do `showToast` at the end?

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -2965,6 +2965,8 @@ en:
           one: "Progress: <strong>%{count}</strong> topic"
           other: "Progress: <strong>%{count}</strong> topics"
         silent: "Perform this action silently."
+        performing: "Performing…"
```
Comment (martin-brennan) [inline config/locales/client.en.yml:2968]:
```suggestion
        performing: "Performing bulk operations, please wait…"
```

minor, I just think this is a tiny bit more descriptive

File: app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs [hunk 3]
```diff
@@ -41,8 +43,8 @@ export default class BulkTopicActions extends Component {
...
+      this.errors = true;
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs:46]: What happens if there is an error in the actual modal? We should probably reset the loading state, form and `this.errors` right?

**Review by martin-brennan — APPROVED** [review]
Looks great, just a couple of minor things and a question about what happens if there is an error :+1:

File: app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs (hunk 1 above)
Comment (oblakeerickson) [reply app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs:45]: Probably!

File: app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs (hunk 3 above)
Comment (oblakeerickson) [reply app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs:46]: Uhh good question. Let me see if I can find a good way to test this.

Comment (oblakeerickson) [reply app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs:45]: https://github.com/discourse/discourse/assets/1490496/bdf2c179-e2b9-48ae-b8b3-9336f1f8a4be

Not really sure how I can do this with a system test. To simulate an error I just edited the controller and forced it to raise an error every time.

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs:45]: All good, I just wanted to see what happened if there was an error and that it was handled correctly :) I am guessing in reality the errors should be rare

**Pull request merged**